tests/adapters/git/test_github_git_provider.py

```diff
@@ -0,0 +1,150 @@
+from unittest import TestCase
+from unittest.mock import MagicMock, patch
+
+from github import AuthenticatedUser, Branch, GithubException
+
+from nestor_api.adapters.git.abstract_git_provider import GitProviderError
+from nestor_api.adapters.git.github_git_provider import GitHubGitProvider
+
+
+@patch("nestor_api.adapters.git.github_git_provider.Github", autospec=True)
+class TestGitHubGitProvider(TestCase):
+    @patch("nestor_api.adapters.git.github_git_provider.Configuration", autospec=True)
+    def test_init(self, configuration_mock, github_mock):
+        """Should init the Github client."""
+        configuration_mock.get_git_provider_token.return_value = "some-token"
+        GitHubGitProvider()
+        github_mock.assert_called_once_with("some-token")
+
+    def test_get_user_info(self, github_mock):
+        """Should get the user data."""
+        fake_user = MagicMock(spec=AuthenticatedUser.AuthenticatedUser)
+        instance = github_mock.return_value
+        instance.get_user.return_value = fake_user
+
+        github_provider = GitHubGitProvider()
+        result = github_provider.get_user_info()
+
+        instance.get_user.assert_called_once()
+        self.assertEqual(result, fake_user)
+
+    def test_get_user_info_failing(self, github_mock):
+        """Should raise a GitProviderError when failing."""
+        instance = github_mock.return_value
+        instance.get_user.side_effect = GithubException(500, "fake error")
+        github_provider = GitHubGitProvider()
+
+        with self.assertRaises(GitProviderError):
+            github_provider.get_user_info()
+
+    def test_get_branch(self, github_mock):
+        """Should get the branch data."""
+        fake_branch_result = MagicMock(spec=Branch.Branch)
+        instance = github_mock.return_value
+        instance.get_repo.return_value.get_branch.return_value = fake_branch_result
+        github_provider = GitHubGitProvider()
+        result = github_provider.get_branch("organization", "fake-project", "fake-branch")
+        instance.get_repo.assert_called_once_with("organization/fake-project")
+        instance.get_repo.return_value.get_branch.assert_called_once_with(branch="fake-branch")
+        self.assertEqual(result, fake_branch_result)
```

Comment:
Could you format the tests this way ? 🙏 🙂 
```suggestion
        fake_branch_result = MagicMock(spec=Branch.Branch)
        instance = github_mock.return_value
        instance.get_repo.return_value.get_branch.return_value = fake_branch_result

        github_provider = GitHubGitProvider()
        result = github_provider.get_branch("organization", "fake-project", "fake-branch")

        instance.get_repo.assert_called_once_with("organization/fake-project")
        instance.get_repo.return_value.get_branch.assert_called_once_with(branch="fake-branch")
        self.assertEqual(result, fake_branch_result)
```